### PR TITLE
Add docs for SYSLIB1230

### PR DESCRIPTION
**Pull request by jkoritzinsky**
## Summary

Describe your changes here.

Contributes to https://github.com/dotnet/docs/issues/42812

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [docs/fundamentals/syslib-diagnostics/source-generator-overview.md](https://github.com/dotnet/docs/blob/a7089ff11862023e6e0e6ca7968f1fd5aae594a2/docs/fundamentals/syslib-diagnostics/source-generator-overview.md) | [docs/fundamentals/syslib-diagnostics/source-generator-overview](https://review.learn.microsoft.com/en-us/dotnet/fundamentals/syslib-diagnostics/source-generator-overview?branch=pr-en-us-42884) |
| [docs/fundamentals/syslib-diagnostics/syslib-cominterfacegenerator.md](https://github.com/dotnet/docs/blob/a7089ff11862023e6e0e6ca7968f1fd5aae594a2/docs/fundamentals/syslib-diagnostics/syslib-cominterfacegenerator.md) | [docs/fundamentals/syslib-diagnostics/syslib-cominterfacegenerator](https://review.learn.microsoft.com/en-us/dotnet/fundamentals/syslib-diagnostics/syslib-cominterfacegenerator?branch=pr-en-us-42884) |
| [docs/fundamentals/syslib-diagnostics/syslib1230.md](https://github.com/dotnet/docs/blob/a7089ff11862023e6e0e6ca7968f1fd5aae594a2/docs/fundamentals/syslib-diagnostics/syslib1230.md) | [docs/fundamentals/syslib-diagnostics/syslib1230](https://review.learn.microsoft.com/en-us/dotnet/fundamentals/syslib-diagnostics/syslib1230?branch=pr-en-us-42884) |

### CODE REVIEW

**Review by AaronRobinsonMSFT — APPROVED** [review]

File: docs/fundamentals/syslib-diagnostics/syslib1230.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
+---
+title: SYSLIB1230 warning
```
[the remark this answers is not in the document]
Comment (jkoritzinsky) [reply docs/fundamentals/syslib-diagnostics/syslib1230.md:2]: This title matches the other documents in this folder.

**Review by jeffhandley — APPROVED** [review]
LGTM - we should wait to merge until we get a review from @gewarren

File: docs/fundamentals/syslib-diagnostics/syslib1090-1099.md [hunk 2]
```diff
@@ -1,5 +1,5 @@
 ---
-title: SYSLIB diagnostics for Microsoft.Interop.COMInteropGenerator
+title: SYSLIB diagnostics for Microsoft.Interop.COMInteropGenerator in .NET 8 and later
```
Comment (gewarren) [inline docs/fundamentals/syslib-diagnostics/syslib1090-1099.md:2]: It's unfortunate that these can't all be in the same file (1090-1099 and 1230-1239). It seems like they should be. Right? It's just the TOC entries that we'd need to figure out.

File: docs/fundamentals/syslib-diagnostics/syslib1230-1239.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+---
+# SYSLIB diagnostics for COM interop source generation in .NET 9 and later
+
+The following table shows the diagnostic IDs for COM interop source-generation analyzers in .NET 9 and later versions.
+
+| Diagnostic ID | Description |
+| - | - |
+| `SYSLIB1230` | Deriving from a `GeneratedComInterface`-attributed interface defined in another assembly is not supported. |
+
+[!INCLUDE [suppress-syslib-warning](includes/suppress-source-generator-diagnostics.md)]
```
Comment (gewarren) [inline docs/fundamentals/syslib-diagnostics/syslib1230-1239.md:25]: If we keep in separate files (which I'm not crazy about) we should at least link between them in the # See also section.

File: docs/fundamentals/syslib-diagnostics/syslib1230.md [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+f1_keywords:
+  - syslib1039
+---
+
+# SYSLIB1230: Deriving from a `GeneratedComInterface`-attributed interface defined in another assembly is not supported.
+
+In .NET 9 and later versions, defining an interface with the <xref:System.Runtime.InteropServices.Marshalling.GeneratedComInterfaceAttribute> attribute that derives from a `GeneratedComInterface`-attributed interface that's defined in another assembly is supported with the following restrictions:
+
+- The base interface type must be compiled targeting the same target framework as the derived type.
```
Comment (gewarren) [inline docs/fundamentals/syslib-diagnostics/syslib1230.md:14]:
```suggestion
- The base interface type must be compiled targeting the same framework as the derived type.
```

File: docs/fundamentals/syslib-diagnostics/syslib1230.md [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+- The base interface type must not shadow any members of its base interface, if it has one.
+
+Additionally, any changes to any generated virtual method offsets in the base interface chain defined in another assembly won't be accounted for in the derived interfaces until the project is rebuilt.
+
+A `SYSLIB1230` warning is emitted when defining an interface with the <xref:System.Runtime.InteropServices.Marshalling.GeneratedComInterfaceAttribute> attribute that derives from a `GeneratedComInterface`-attributed interface that's defined in another assembly to inform you of the limitations. To acknowledge the limitations, suppress the warning in code as follows:
```
Comment (gewarren) [inline docs/fundamentals/syslib-diagnostics/syslib1230.md:19]:
```suggestion
When you define an interface with the <xref:System.Runtime.InteropServices.Marshalling.GeneratedComInterfaceAttribute> attribute that derives from a `GeneratedComInterface`-attributed interface that's defined in another assembly, the SYSLIB1230 warning is emitted to inform you of the limitations. To acknowledge the limitations, suppress the warning in code as follows:
```

File: docs/fundamentals/syslib-diagnostics/syslib1090-1099.md (hunk 2 above)
Comment (jkoritzinsky) [reply docs/fundamentals/syslib-diagnostics/syslib1090-1099.md:2]: Yeah I don't like them being in separate files. I was just following the current pattern. I got lucky that we ended up splitting the diagnostic groups by release version.

I'm fine changing this to however you'd like to make it one file; I just didn't want to make assumptions about how you'd want to do it.

File: docs/fundamentals/syslib-diagnostics/syslib1230.md [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+{
+    // the interface definition
+}
+```
+
+[!INCLUDE [suppress-syslib-warning](includes/suppress-source-generator-diagnostics.md)]
+
+## See also
+
+- [SYSLIB diagnostics for COM interop source generation](syslib-cominteropgenerator.md)
```
Comment (gewarren) [inline docs/fundamentals/syslib-diagnostics/syslib1230.md:36]:
```suggestion
- [SYSLIB diagnostics for COM interop source generation](syslib-cominterfacegenerator.md)

```

File: docs/fundamentals/syslib-diagnostics/source-generator-overview.md [hunk 7]
```diff
@@ -253,16 +254,16 @@ The following table provides an index to the `SYSLIB1XXX` diagnostics in .NET 6
...
+[1090]: syslib-cominteropgenerator.md
+[1091]: syslib-cominteropgenerator.md
+[1092]: syslib-cominteropgenerator.md
+[1093]: syslib-cominteropgenerator.md
+[1094]: syslib-cominteropgenerator.md
+[1095]: syslib-cominteropgenerator.md
+[1096]: syslib-cominteropgenerator.md
+[1097]: syslib-cominteropgenerator.md
+[1098]: syslib-cominteropgenerator.md
+[1099]: syslib-cominteropgenerator.md
```
Comment (gewarren) [inline docs/fundamentals/syslib-diagnostics/source-generator-overview.md:266]:
```suggestion
[1090]: syslib-cominterfacegenerator.md
[1091]: syslib-cominterfacegenerator.md
[1092]: syslib-cominterfacegenerator.md
[1093]: syslib-cominterfacegenerator.md
[1094]: syslib-cominterfacegenerator.md
[1095]: syslib-cominterfacegenerator.md
[1096]: syslib-cominterfacegenerator.md
[1097]: syslib-cominterfacegenerator.md
[1098]: syslib-cominterfacegenerator.md
[1099]: syslib-cominterfacegenerator.md
```

File: docs/fundamentals/syslib-diagnostics/syslib1230.md [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+    // the interface definition
+}
+```
+
+[!INCLUDE [suppress-syslib-warning](includes/suppress-source-generator-diagnostics.md)]
+
+## See also
+
+- [SYSLIB diagnostics for COM interop source generation](syslib-cominterfacegenerator.md)
+
```
Comment (gewarren) [inline docs/fundamentals/syslib-diagnostics/syslib1230.md:37]:
```suggestion
```

**Pull request merged**